exonum/src/proto/mod.rs

```diff
@@ -335,6 +339,86 @@ where
     }
 }
 
+impl<V> ProtobufConvert for exonum_merkledb::ListProof<V>
+where
+    V: BinaryValue,
+{
+    type ProtoStruct = ListProof;
+
+    fn to_pb(&self) -> Self::ProtoStruct {
+        let mut list_proof = ListProof::new();
+        list_proof.set_length(self.list_len());
+
+        let entries = self
+            .entries_unchecked()
+            .iter()
+            .map(|(index, value)| {
+                let mut entry = ListProofEntry::new();
+                entry.set_index(*index);
+                entry.set_value(value.to_bytes());
+                entry
+            })
+            .collect();
+
+        let proof = self
+            .proof_unchecked()
+            .iter()
+            .map(|entry| entry.into())
+            .collect();
+
+        list_proof.set_proof(RepeatedField::from_vec(proof));
+        list_proof.set_entries(RepeatedField::from_vec(entries));
+
+        list_proof
+    }
+
+    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error> {
+        let proof = pb
+            .get_proof()
+            .iter()
+            .map(|entry| {
+                Ok((
+                    exonum_merkledb::ProofListKey::new(
```

Comment:
It is necessary to check ranges for `height` and `index`. The constructor will crash if `index > 1 << 58` with the debug assertions on, and probably can manifest unintended behavior further on if the assertions are off.